core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -704,7 +704,13 @@ private[spark] object PythonRunner {
   // already running worker monitor threads for worker and task attempts ID pairs
   val runningMonitorThreads = ConcurrentHashMap.newKeySet[(Socket, Long)]()
 
+  private var printPythonInfo = true
+
   def apply(func: PythonFunction): PythonRunner = {
+    if (printPythonInfo) {
+      printPythonInfo = false
+      PythonUtils.logPythonInfo(func.pythonExec)
+    }
```

Comment:
Is this the only way? It looks like too much to check `if` condition at every `apply` invocation.

Reply:
Yes, this is the first place, I could see in Executor, where we get the expected Python program name from the driver

Reply:
Dongjoon has more context around whether this is the right place.
One comment - if it is indeed the only place to put this, then `printPythonInfo` is being used in an MT-unsafe way.
Make it an `AtomicBoolean`